src/redprl/refiner.fun

```diff
@@ -1015,8 +1013,190 @@ struct
       end
   end
 
+  structure Restriction =
+  struct
+    (* This structure provides functions that automate the restriction
+       judgement rules given in "Dependent Cubical Realizability",
+       page 46.
+
+       Such automation is possible because the rules are "syntax
+       directed" on the restriction being performed.
```

Comment:
great! This is just the sort of thing that Carlo and I had envisioned, and thanks for putting the reference here.